File: Game.Application/GameJS/src/utils/specifications/OrSpecification.ts
import { ISpecification } from "./ISpecification";
export class OrSpecification<TSituation> implements ISpecification<TSituation, boolean> {
  private readonly original: Array<ISpecification<TSituation, boolean>>;
  constructor(...original: Array<ISpecification<TSituation, boolean>>) {
    this.original = original;
  }
  execute(situation: TSituation): boolean {
    return this.original.some(spec => spec.execute(situation));
  }
}

File: Game.Domain/Specifications/OrSpecification.cs
using System.Linq;

namespace Game.Domain.Specifications
{
    public class OrSpecification<TSituation> : ISpecification<TSituation, bool>
    {
        private readonly ISpecification<TSituation, bool>[] original;

        public OrSpecification(params ISpecification<TSituation, bool>[] original)
        {
            this.original = original;
        }

        public bool Execute(TSituation situation)
        {
            return original.Any(s => s.Execute(situation));
        }
    }

}
